app/models/robots/bed/pooling.rb

```diff
@@ -7,23 +7,45 @@ class Pooling < Robots::Bed::Base
 
     def each_parent
       range.each do |i|
-        plate_barcode = parent_plates[i]&.barcode || SBCF::EmptyBarcode.new
-        yield(parents[i], plate_barcode)
+        labware_barcode = parent_labwares[i]&.barcode || SBCF::EmptyBarcode.new
+        yield(parents[i], labware_barcode)
       end
     end
 
+    def load(barcodes)
+      # Ensure we always deal with an array, and any accidental duplicate scans are squashed out
+      @barcodes = Array(barcodes).map(&:strip).uniq.reject(&:blank?)
+
+      @labwares = if @barcodes.present?
+                    Sequencescape::Api::V2::Plate.find_all({ barcode: @barcodes }, includes: labware_includes)
+                  else
+                    []
+                  end
+    end
+
     private
 
-    def parent_plates
-      @parent_plates ||= plate.wells.sort_by(&well_order).each_with_object([]) do |well, plates|
+    def parent_labwares
+      return [] if labware.nil?
+
+      @parent_labwares ||= if labware.plate?
+                             parent_labwares_of_plate
+                           else
+                             # currently not used for tubes
+                             []
```

Comment:
Isn't this needed for tubes? How are you validating that a tube is paired with the correct plate?